src/sql/workbench/services/objectExplorer/browser/objectExplorerService.ts

```diff
@@ -407,67 +409,92 @@ export class ObjectExplorerService implements IObjectExplorerService {
 				}
 				let provider = this._providers[providerId];
 				if (provider) {
-					let resultMap: Map<string, azdata.ObjectExplorerExpandInfo> = new Map<string, azdata.ObjectExplorerExpandInfo>();
-					let allProviders: azdata.ObjectExplorerProviderBase[] = [provider];
+					try {
+						let resultMap: Map<string, azdata.ObjectExplorerExpandInfo> = new Map<string, azdata.ObjectExplorerExpandInfo>();
+						let allProviders: azdata.ObjectExplorerProviderBase[] = [provider];
+
+						let nodeProviders = this._nodeProviders[providerId];
+						if (nodeProviders) {
+							nodeProviders = nodeProviders.sort((a, b) => a.group!.toLowerCase().localeCompare(b.group!.toLowerCase()));
+							allProviders.push(...nodeProviders);
+						}
 
-					let nodeProviders = this._nodeProviders[providerId];
-					if (nodeProviders) {
-						nodeProviders = nodeProviders.sort((a, b) => a.group!.toLowerCase().localeCompare(b.group!.toLowerCase()));
-						allProviders.push(...nodeProviders);
-					}
 
-					self._sessions[session.sessionId!].nodes[node.nodePath].expandEmitter.event((expandResult: NodeExpandInfoWithProviderId) => {
-						if (expandResult && expandResult.providerId) {
-							resultMap.set(expandResult.providerId, expandResult);
-							// If we got an error result back then send error our error event
-							// We only do this for the MSSQL provider
-							if (expandResult.errorMessage && expandResult.providerId === mssqlProviderName) {
-								const errorType = expandResult.errorMessage.indexOf('Object Explorer task didn\'t complete') !== -1 ? 'Timeout' : 'Other';
-								// For folders send the actual name of the folder (since the nodeTypeId isn't useful in this case and the names are controlled by us)
-								const nodeType = node.nodeTypeId === NodeType.Folder ? node.label : node.nodeTypeId;
-								this._telemetryService.createErrorEvent(TelemetryKeys.TelemetryView.Shell, TelemetryKeys.TelemetryError.ObjectExplorerExpandError, undefined, errorType)
-									.withAdditionalProperties({
-										nodeType,
-										providerId: expandResult.providerId
-									}).send();
+
+						let retryCount = 0;
+						const expansionTimeout = this._configurationService.getValue<number>('serverTree.nodeExpansionTimeout');
+						const timeout = setTimeout(() => {
+							/**
+							 * If we don't get a response back from the provider in specified expansion timeout seconds then we assume
+							 * it's not going to respond and resolve the promise with the results we have so far
+							 */
+							if (resultMap.size === allProviders.length || retryCount === expansionTimeout) {
```

Comment:
We should definitely be informing the user when a timeout is hit - toast notification is probably the easiest but could consider an error node as well similar to the current MSSQL experience.